feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/domain/StakingInteractor.kt

```diff
@@ -256,12 +257,18 @@ class StakingInteractor(
         val networkType = state.accountAddress.networkType()
         val tokenType = Token.Type.fromNetworkType(networkType)
 
+        val totalRewardsResult =
+            when (networkType) {
+                Node.NetworkType.KUSAMA -> tokenType.amountFromPlanks(stakingRewardsRepository.stakingRewardSubQueryKusama(state.accountAddress))
+                Node.NetworkType.POLKADOT -> tokenType.amountFromPlanks(stakingRewardsRepository.stakingRewardSubQueryPolkadot(state.accountAddress))
+                Node.NetworkType.WESTEND -> totalRewards(stakingRewardsRepository.stakingRewardsFlow(state.accountAddress).first())
+            }
+
         combine(
             stakingRepository.electionFlow(networkType),
             stakingRepository.observeActiveEraIndex(networkType),
-            stakingRewardsRepository.stakingRewardsFlow(state.accountAddress),
             walletRepository.assetFlow(state.accountAddress, tokenType)
-        ) { electionStatus, activeEraIndex, rewards, asset ->
+        ) { electionStatus, activeEraIndex, asset ->
```

Comment:
We cannot just throw away flow here, since database is filled not atomically for Subscan-way, but rather it is updated for each page retrieved. Thus, we need to deliver this updates to UI too

Also, i do not see how subquery value is cached 

My suggestion is to have separate local datasource (aka cache) for subquery, which will be simple StateFlow with initial value being retrieved from shared preferences, where it is written each time the successfull sync happens